src/tock_genai_core/utils/kube/kube_secret_manager_client.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+"""
+KubeSecretManagerClient
+
+Minimal Kubernetes Secret Manager client.
+Used to fetch secrets from a Kubernetes cluster.
+
+Author:
+    * Adapted for kube integration
+"""
+
+import base64
+import json
+import logging
+from typing import Union
+
+from kubernetes import client, config
+from kubernetes.client.exceptions import ApiException
+
+logger = logging.getLogger(__name__)
+
+
+class KubeSecretManagerClient:
+    """Minimal Kubernetes Secret Manager Client."""
+
+    def __init__(self, namespace: str = "default"):
+        """Initialize client
+
+        Args:
+            namespace (str, optional): Namespace from which secrets will be retrieved. Defaults to "default".
+        """
+
+        try:
+            config.load_incluster_config()
+            logger.info("Using in-cluster Kubernetes configuration.")
+        except Exception:
+            config.load_kube_config()
+            logger.info("Using local kubeconfig.")
+        self.api = client.CoreV1Api()
+        self.namespace = namespace
+
+    def get_secret(self, secret_name: str) -> Union[str, dict, None]:
+        """Retrieve an individual secret from Kubernetes.
```

Comment:
Explain in the comment that it will try to load it as json
```suggestion
        """Retrieve an individual secret from Kubernetes. Will try to parse it as JSON.
```